File: src/main/scala/com/cotter/PbIntDeserializer.scala
package com.cotter

import java.util

import com.cotter.io.models.SimpleMessages.SimpleInt
import org.apache.kafka.common.serialization.Deserializer

class PbIntDeserializer extends Deserializer[SimpleInt] {
  override def deserialize(topic: String, bytes: Array[Byte]): SimpleInt = SimpleInt.parseFrom(bytes)

  override def configure(configs: util.Map[String, _], isKey: Boolean): Unit = ()

  override def close(): Unit = ()
}

File: service/src/main/java/com/jxc454/histoservice/PbIntDeserializer.java
package com.jxc454.histoservice;

import com.jxc454.models.SimpleMessages.SimpleInt;
import com.jxc454.models.shaded.protobuf.InvalidProtocolBufferException;
import org.apache.kafka.common.serialization.Deserializer;

public class PbIntDeserializer implements Deserializer<SimpleInt> {
  public SimpleInt deserialize(String topic, byte[] data){
      try {
          return SimpleInt.parseFrom(data);
      } catch (final InvalidProtocolBufferException e) {
          throw new RuntimeException("Received unparseable message " + e.getMessage(), e);
      }
  }

  public void configure(java.util.Map<String, ?> configs, boolean isKey) {}

  public void close() {}
}
